cmd/clusterctl/client/repository/repository_github.go

```diff
@@ -107,7 +107,7 @@ func (g *gitHubRepository) GetFile(version, path string) ([]byte, error) {
 }
 
 // newGitHubRepository returns a gitHubRepository implementation
-func newGitHubRepository(providerConfig config.Provider, configVariablesClient config.VariablesClient) (*gitHubRepository, error) {
+func newGitHubRepository(providerConfig config.Provider, configVariablesClient config.VariablesClient, injectClient *github.Client) (*gitHubRepository, error) {
```

Comment:
Could we perhaps use an options pattern here? Something like:

```Golang
type GithubRepositoryOption func(*gitHubRepository)

func InjectGithubClient(c *github.Client) GithubRepositoryOption {
  return func(g *gitHubRepository) {
    g.injectClient = c
  }
}

func newGitHubRepository(
  providerConfig config.Provider, 
  configVariablesClient config.VariablesClient, 
  opts ...GithubRepositoryOption
) (*gitHubRepository, error)
```

This way since the last arg is variadic we don't need to change the function signature everywhere else it isn't needed.

Reply:
yep, all of these make sense. fixed in amended commit.